### ISSUE TITLE
Features extraction

### ISSUE THREAD

**Issue by mab85** [timzhang642/DCGAN-Image-Generation#1]
@timzhang642 I want to use discriminator network to extract features of my dataset. My question is how decide to stop training dcgan and take the features?. In other meaning what the criteria for stopping training dcgan?. Recently some papers use early stopping method but already dcgan do not suffer from over-fitting because we aren't able to find the Nash equilibrium very well [https://arxiv.org/pdf/1611.04273.pdf](url)

**Comment by timzhang642** [comment]
I have not tried this but I'd say stop training the discriminator when you see the loss for both generator and discriminator converge.

**Comment by mab85** [comment]
@timzhang642 How can stop training depends on train loss and val loss after each training epoch?

**Comment by TanmDL** [comment]
Thank you for sharing the code. I have a similar question on feature extractions. For extracting features,  we have to apply max pooling on  dis1, dis2, dis3 and then concatenate it so that it will become a 2D vector (batch size, concatenate features).  
Is it so ? or am I missing something?
